Scripts/Process/ProcessContainer.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace VRTK.Core.Process
+{
+    using UnityEngine;
+
+    /// <summary>
+    /// The ProcessContainer acts as a container for a process that can be utilized within a Unity Inspector.
+    /// </summary>
+    public abstract class ProcessContainer

```

Comment:
This is generically usable for **any** interface and as such I think we should change all naming (and docs!) to read `InterfaceContainer` instead. If we do so we need to remember to update the `PropertyDrawer` for it, too.

Reply:
if it's not just for `process`es then what about the `protected Object process` field? should that be called something other than `process`?

Reply:
Yup, let's call it `target` perhaps?